src/index.ts

```diff
@@ -0,0 +1,101 @@
+import { TASK_FLATTEN_GET_FLATTENED_SOURCE } from "@nomiclabs/buidler/builtin-tasks/task-names";
+import { extendEnvironment, task } from "@nomiclabs/buidler/config";
+import { BuidlerPluginError, lazyObject } from "@nomiclabs/buidler/plugins";
+
+import AbiEncoder from "./AbiEncoder";
+import ContractCompiler from "./ContractCompiler";
+import EtherscanService from "./etherscan/EtherscanService";
+import EtherscanVerifyContractRequest from "./etherscan/EtherscanVerifyContractRequest";
+import SolcVersions from "./solc/SolcVersions";
+
+export class EtherscanBuidlerEnvironment {
+  constructor(
+    public readonly url: string = "https://api.etherscan.io/api",
+    public readonly token: string = ""
+  ) {}
+}
+
+declare module "@nomiclabs/buidler/types" {
+  export interface BuidlerRuntimeEnvironment {
+    etherscan: EtherscanBuidlerEnvironment;
+  }
+
+  export interface ResolvedBuidlerConfig {
```

Comment:
You should extend BuidlerConfig
```
export interface EtherscanConfig extends BuidlerConfig {
  etherscan?: {
    url?: string;
    token?: string;
  }
}

```

Reply:
Hmm I'm actually redeclaring `ResolvedBuidlerConfig` to effectively extend it so typescript compiler doesn't complain when I'm using values provided by `buidler.config.js`.

`env.config` [here](https://github.com/nomiclabs/buidler-etherscan/pull/2/files/f1c0b7782d12c2d630263f4082bfe8d4e46cabab#diff-f41e9d04a45c83f3b6f6e630f10117feR39) is of type `ResolvedBuidlerConfig` and without redeclaration it wouldn't know about etherscan props.

Reply:
There's been some confusion here. You do need to redeclare a config type, but that's `BuidlerConfig`, not `ResolvedBuidlerConfig`. Note that everything has to be optional, as that's the user-defined config's type.

Something like this should work

```ts
export interface BuidlerConfig {
  etherscan?: {
    url?: string;
    token?: string;
  }
}
```

`ResolvedBuidlerConfig` extends `BuidlerConfig` so you won't have problems with `env.config`.

Reply:
makes sense, will update